modules/edge-identity/src/identity.rs

```diff
@@ -156,17 +158,53 @@ decl_module! {
 				IdentityStage::Attested => ()
 			}
 
+			// Check the number of verifications the record has and ensure all are valid
+			let mut valid_verifications: Vec<(T::AccountId, bool)> = [(_sender.clone(), vote)].to_vec();
+			if let Some(verifications) = record.verifications {
+				valid_verifications = verifications.clone()
+					.into_iter()
+					.filter(|v| Self::verifiers().contains(&v.0))
+					.collect();
+				valid_verifications.push((_sender.clone(), vote));
+			}
+
+			let yes_votes: Vec<(T::AccountId, bool)> = valid_verifications.clone()
+				.into_iter()
+				.filter(|v| v.1 == true)
+				.collect();
+
+			let no_votes: Vec<(T::AccountId, bool)> = valid_verifications.clone()
+				.into_iter()
+				.filter(|v| v.1 == false)
+				.collect();
+
+			// Check if we have gathered a supermajority of yes votes
+			let mut stage = IdentityStage::Attested;
+			let mut expiration_time = record.expiration_time;
+			if yes_votes.len() * 3 >= 2 * Self::verifiers().len() {
+				stage = IdentityStage::Verified;
+				expiration_time = None;
+			}
+
+			// Check if we have gather a supermajority of no votes
+			// TODO: What does it mean to fail a verification? If malicious
+			//		 behavior, we should punish the sending account somehow.
+			if no_votes.len() * 3 >= 2 * Self::verifiers().len() {
+				Self::remove_pending_identity(&identity_hash);
```

Comment:
We should emit an event here if we're going to do this.

Reply:
Also we'd need a "failed" stage or something.

Reply:
Good idea.